### PR TITLE
[Form] ignore microseconds submitted by Edge

### LINKED ISSUE #35989: [Form] TimeType does not work in new Edge
**Symfony version(s) affected**: all

**Description**  
When submitting a form via Edge using the `TimeType`; Edge will submit the value with both seconds and milliseconds every time regardless of the `step` attribute (e.g. `20:11:00.000`). This is valid [according to W3C](https://www.w3.org/TR/2012/WD-html-markup-20120329/input.time.html).

It looks like Symfony is unable to parse the milliseconds in the `DateTimeToStringTransformer` and this then causes an invalid value.

This only seems to be an issue since Edge has moved to Webkit.

**How to reproduce**  

1. Create a form with a TimeType
2. Attempt to submit the form with a value in Edge 80+

**Possible Solution**  

**Additional context**

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #35989
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/Symfony/Component/Form/Extension/Core/Type/TimeType.php [hunk 1]
```diff
@@ -57,16 +57,18 @@ public function buildForm(FormBuilderInterface $builder, array $options)
...
-            // handle seconds ignored by user's browser when with_seconds enabled
-            // https://codereview.chromium.org/450533009/
-            if ($options['with_seconds']) {
-                $builder->addEventListener(FormEvents::PRE_SUBMIT, function (FormEvent $e) {
-                    $data = $e->getData();
-                    if ($data && preg_match('/^\d{2}:\d{2}$/', $data)) {
-                        $e->setData($data.':00');
+            $builder->addEventListener(FormEvents::PRE_SUBMIT, function (FormEvent $e) use ($options) {
+                $data = $e->getData();
+                if ($data && preg_match('/^(?P<hours>\d{2}):(?P<minutes>\d{2})(?::(?P<seconds>\d{2})\.0{3})?$/', $data, $matches)) {
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Form/Extension/Core/Type/TimeType.php]: do we need to support missing milliseconds?
https://tools.ietf.org/html/rfc3339#page-8 looks like the end could be `(?:\.0{1,2})?`?

Comment (xabbuh) [reply src/Symfony/Component/Form/Extension/Core/Type/TimeType.php]: I didn't account for that as I understood that Edge would always send milliseconds. But being even less restrictive might be a good idea to account for further changes in browsers.

Comment (xabbuh) [reply src/Symfony/Component/Form/Extension/Core/Type/TimeType.php]: code updated and test added

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @xabbuh.

**Pull request merged**